cantools/database/utils.py

```diff
@@ -79,8 +79,8 @@ def _encode_fields(fields: Sequence[Union["Signal", "Data"]],
             unpacked[field.name] = _transform(value)  # type: ignore[operator]
             continue
 
-        if isinstance(value, Decimal):
-            unpacked[field.name] = (value - field.decimal.offset) / field.decimal.scale
+        if isinstance(value, Decimal) and isinstance(field, Signal):
```

Comment:
I am not very happy with the second check, but I guess we need it since class `Data` does not have the `decimal` attribute. I don't see easier way to implement this